### ISSUE TITLE
WW3 ICs are not read for HR like experiements. 

### ISSUE THREAD

**Issue by JessicaMeixner-NOAA** [NOAA-EMC/global-workflow#3109]
### What is wrong?

The gfs_stage_ic job copies wave ICs: 

` ^[[38;21m2024-11-17 13:45:38,610 - INFO     - file_utils  : Copied /lfs/h2/emc/couple/noscrub/jessica.meixner/WaveUglo15km/ICDIR/Opt3/gfs.20200912/18/model/wave/restart/20200913.000000.restart.ww3 to /lfs/h2/emc/couple/noscrub/jessica.meixner/WaveUglo15km/Opt3Sept/COMROOT/Opt3Sept/gfs.20200912/18//model/wave/restart^[[0m`

But then the forecast job does not find ICs: 
```
+ exglobal_forecast.sh[124]: WW3_postdet
+ forecast_postdet.sh[327]: echo 'SUB WW3_postdet: Linking input data for WW3'
SUB WW3_postdet: Linking input data for WW3
+ forecast_postdet.sh[329]: local ww3_grid first_ww3_restart_out ww3_restart_file
+ forecast_postdet.sh[331]: [[ .false. == \.\t\r\u\e\. ]]
+ forecast_postdet.sh[357]: echo 'WW3 will start from rest!'
WW3 will start from rest!
+ forecast_postdet.sh[358]: first_ww3_restart_out=2020091300
+ forecast_postdet.sh[362]: local ww3_restart_file
+ forecast_postdet.sh[364]: (( vdate = first_ww3_restart_out ))
+ forecast_postdet.sh[364]: (( vdate <= forecast_end_cycle ))
+ forecast_postdet.sh[366]: ww3_restart_file=20200913.000000.restart.ww3
+ forecast_postdet.sh[367]: /bin/ln -sf /lfs/h2/emc/stmp/jessica.meixner/RUNDIRS/Opt3Sept/gfs.2020091300/gfsfcst.2020091300/restart/WW3_RESTART/20200913.000000.restart.ww3 20200913.000000.restart.ww3
```

### What should have happened?

ICs should be used in the HR prototype-like cases.   

### What machines are impacted?

All or N/A

### What global-workflow hash are you using?

Techncially it's a branch, but this also appears to have happend in HR4 tag 

### Steps to reproduce

I'm running my branch here: https://github.com/JessicaMeixner-NOAA/global-workflow/tree/feature/uglo_15km 
and staging my own IC directory. 

So from: /lfs/h2/emc/couple/noscrub/jessica.meixner/WaveUglo15km/global-workflow/workflow 
using the script coupled.sh to set up experiments. 

### Additional information

I checked HR4 output and there are no wave ICs as HS is 0 at the first output (haven't confirmed from log though). 

Have pinged @sbanihash to be sure to check GEFS recent experiment results.  

Have not yet checked low-res Ci tests, but there's likely an example use-case that could be used for debugging. 

### Do you have a proposed solution?

Trying to figure out logic https://github.com/NOAA-EMC/global-workflow/blob/develop/ush/forecast_postdet.sh#L330-L359 and in the stage_ic job to figure out where the disconnect is.

**Comment by JessicaMeixner-NOAA** [comment]
It appears that C48_S2SW CI test will demonstrate the issue.

**Comment by JessicaMeixner-NOAA** [comment]
Okay - so running the C48_S2SW CI test can replicate this issue some log files here: 
/scratch1/NCEPDEV/climate/Jessica.Meixner/WaveICIssue/test01/COMROOT/test01/logs/2021032312 

Stage_IC does: 

`^[[38;21m2024-11-18 19:43:29,513 - INFO     - file_utils  : Copied /scratch1/NCEPDEV/global/glopara/data/ICSDIR/C48mx500/20240610/gfs.20210323/06/model/wave/restart/20210323.120000.restart.uglo_100km to /scratch1/NCEPDEV/climate/Jessica.Meixner/WaveICIssue/test01/COMROOT/test01/gfs.20210323/06//model/wave/restart/20210323.120000.restart.ww3^[[0m`


Fcst job goes into this section, it's a cold start and then just simply states: 
https://github.com/NOAA-EMC/global-workflow/blob/develop/ush/forecast_postdet.sh#L356-L359
"echo "WW3 will start from rest!"" 

However, many times we have staged IC, so I don't quite understand this logic or when this changed happened.  Still trying to go back and understand that. 

@aerorahul @WalterKolczynski-NOAA @KateFriedman-NOAA or others are you aware of why this is doing this? 

Some PRs of potential reference: 
https://github.com/NOAA-EMC/global-workflow/pull/2510
https://github.com/NOAA-EMC/global-workflow/pull/3009 
And the changes in stage_ic. 

For GEFS @sbanihash @NeilBarton-NOAA  is there a CI test that has staged wave ICs that I can run to make sure I don't break that with these updates.

**Comment by NeilBarton-NOAA** [comment]
@JessicaMeixner-NOAA The GEFS CI test is at https://github.com/NOAA-EMC/global-workflow/blob/develop/ci/cases/pr/C96_S2SWA_gefs_replay_ics.yaml

**Comment by WalterKolczynski-NOAA** [comment]
It was changed in #2510 when Rahul overhauled the entire set of forecast scripts. I think I actually fixed it in #3009, but neglected to remove the (I think) now-erroneous message.

**Comment by JessicaMeixner-NOAA** [comment]
> It was changed in #2510 when Rahul overhauled the entire set of forecast scripts. I think I actually fixed it in #3009, but neglected to remove the (I think) now-erroneous message.

@WalterKolczynski-NOAA  - where are the ICs copied then?  I  checked the actual output of the wave model and there is no wave IC being used, so it's not just an erroneous error message -- this is a huge problem.  In discussion with @NeilBarton-NOAA @sbanihash - we think it'd be a good idea to let the user chose if they want to error out if there is no wave IC or not and control that by a flag so that people can perhaps have an easier way of knowing about this.   I think GEFS is okay b/c it's a warm start versus the tests I'm using which are cold starts which put you in different parts of the loop, but we'll continue to check.

**Comment by WalterKolczynski-NOAA** [comment]
Okay, I see the problem now. Should be an easy fix.

**Comment by JessicaMeixner-NOAA** [comment]
@WalterKolczynski-NOAA  - I've been working on a fix for this, however it'll be great if you take this over. If you will be working on this, can you let me know what your tlimeline for a fix will be?

**Comment by WalterKolczynski-NOAA** [comment]
> @WalterKolczynski-NOAA - I've been working on a fix for this, however it'll be great if you take this over. If you will be working on this, can you let me know what your tlimeline for a fix will be?

I'm going to do it today.

**Comment by JessicaMeixner-NOAA** [comment]
@WalterKolczynski-NOAA  great thanks! 

Just to share where I was at, I have local changes here: /scratch1/NCEPDEV/climate/Jessica.Meixner/WaveICIssue/global-workflow which I tested here: /scratch1/NCEPDEV/climate/Jessica.Meixner/WaveICIssue/test02 and they did not work because this did not match the stage_ic location.  I'm not sure if this means that the stage IC job should also get updated to be more in sync w/where a "warm" start IC would be expected or if I did not update things correctly on my end or made another mistake -but thought I'd share where I was at.
